packages/gems/js/ext/js/js-core.c

```diff
@@ -200,6 +200,9 @@ static VALUE _rb_js_obj_aref(VALUE obj, VALUE key) {
 static VALUE _rb_js_obj_aset(VALUE obj, VALUE key, VALUE val) {
   struct jsvalue *p = check_jsvalue(obj);
   VALUE rv = _rb_js_try_convert(rb_mJS, val);
+  if (rv == Qnil) {
+    rb_raise(rb_eTypeError, "value is not a JS::Object like object");
```

Comment:
Can we include key name in the error message?

Reply:
Not impossible, but how exactly would you modify the message? Can you provide an example?

Reply:
Ah, sorry I was wrong. I wanted to say not key name but the actual original value and expected type. Also please follow the conventional statement widely used in Ruby standard library like "wrong argument type Foo (expected JS::Object like object)"